proto/one/enterprise/data/enterprise_dataset_definition.proto

```diff
@@ -24,6 +24,9 @@ message EnterpriseDatasetDefinition {
 
     // One or more DataTables that composes the Dataset
     repeated EnterpriseDataTableDefinition dataTables = 6;
+    
+    // DataSource query parameters
+    repeated EnterpriseDataTableQueryParameter queryParameters = 7;
```

Comment:
I have a couple questions about this.

1. How will you know which dataTable in the dataSet each queryParameter belongs to?
2. EnterpriseDataTableDefinition has an EnterpriseDataTableQuery which has a list of EnterpriseDataTableQueryParameters so why do we need a separate list of queryParameters associated directly to the dataSet?
3. If this is DataSource queryParameters as the comment indicates then why not add this field to the EnterpriseDataSourceDefinition model?

Reply:
@AQI-DanG , good questions, and thanks to bring it up.

The new dataset level query parameter will be cascaded to the DataTables level query parameters, it will overwrite and combine with datatable queryparameters together then pass to datasource drivers.

When report/dashboard ask dataset, it will pass down a payload (e.g. time range) to our API, the API will load the dataset and pass payload into dataset level query parameters and proceed the per table data retrieval.

You can find more detail at the confluence page:
https://aquaticinformatics.atlassian.net/wiki/spaces/AOS/pages/254378121/Enterprise+Data+DataSetDefinition#Query-Parameters

Reply:
Thanks for the comment explaining things @AlanLiu-AI .  That makes sense now.  The only suggested change would be a little more description on the comment on the field so it is more obvious what it is for.

Reply:
Comments has been updated, and thanks for the questions and suggestions.